core/src/test/java/io/grpc/ForwardingChannelBuilderTest.java

```diff
@@ -32,11 +31,29 @@
  * Unit tests for {@link ForwardingChannelBuilder}.
  */
 @RunWith(JUnit4.class)
-public class ForwardingChannelBuilderTest {
+public class ForwardingChannelBuilderTest extends AbstractForwardingTest<ManagedChannelBuilder<?>> {
   private final ManagedChannelBuilder<?> mockDelegate = mock(ManagedChannelBuilder.class);
 
   private final ForwardingChannelBuilder<?> testChannelBuilder = new TestBuilder();
 
+  @Override
+  public ManagedChannelBuilder<?> mockDelegate() {
+    return mockDelegate;
+  }
+
+  @Override
+  public ManagedChannelBuilder<?> forwarder() {
+    return testChannelBuilder;
+  }
+
+  @Override
+  public Class<ManagedChannelBuilder<?>> delegateClass() {
```

Comment:
If you are a super cool turbo hacker, you can actually extract the class from the generics on line 34, which allows you to move this method in the the parent.

Reply:
TIL